libp2p/Session.cpp

```diff
@@ -275,8 +275,22 @@ void Session::drop(DisconnectReason _reason)
 	m_peer->m_lastDisconnect = _reason;
 	if (_reason == BadProtocol)
 	{
-		m_peer->m_rating /= 2;
-		m_peer->m_score /= 2;
+		int old_rate = 0;
```

Comment:
Could we extract this to a separate helper function e.g. `halveAtomicInt(std::atomic<int>&)` ?

Also naming convention prescribes `oldRate` & `newRate`